Pod/Classes/CDTQQueryValidator.m

```diff
@@ -138,6 +149,80 @@ + (NSArray *)addImplicitEq:(NSArray *)andClause
     return [NSArray arrayWithArray:accumulator];
 }
 
++ (NSArray *)handleShortHandOperators:(NSArray *)clause
+{
+    NSMutableArray *accumulator = [NSMutableArray array];
+    
+    for (NSDictionary *fieldClause in clause) {
+        NSObject *predicate = nil;
+        NSString *fieldName = nil;
+        if ([fieldClause isKindOfClass:[NSDictionary class]] && [fieldClause count] != 0) {
+            fieldName = fieldClause.allKeys[0];
+            predicate = fieldClause[fieldName];
+            if ([fieldName hasPrefix:@"$"] && [predicate isKindOfClass:[NSArray class]]) {
+                predicate = [CDTQQueryValidator handleShortHandOperators:(NSArray *) predicate];
+            } else if ([predicate isKindOfClass:[NSDictionary class]] &&
+                       [(NSDictionary *)predicate count] != 0) {
+                // if the clause isn't a special clause (the field name starts with
+                // $, e.g., $and), we need to check whether the clause has a shorthand
+                // operator like $ne. If it does, we need to convert it to its longhand
+                // version.
+                // Take:  { "$ne" : ... }
+                // Make:  { "$not" : { "$eq" : ... } }
+                predicate = [CDTQQueryValidator replaceWithLonghand:(NSDictionary *)predicate];
+            } else {
+                [accumulator addObject:fieldClause];
+                continue;
+            }
+        } else {
+            // if this isn't a dictionary, we don't know what to do so add the clause
+            // to the accumulator to be dealt with later as part of the final selector
+            // validation.
+            [accumulator addObject:fieldClause];
+            continue;
+        }
+        
+        [accumulator addObject:@{fieldName : predicate}];  // can't put nil in this
+    }
+    
+    return [NSArray arrayWithArray:accumulator];
+}
+
+/**
+ * This method traverses the predicate dictionary and once it reaches the last operator
+ * in the tree, it checks it for a shorthand representation.  If one exists then
+ * that shorthand representation is replaced with its longhand version.
+ * For example:   { "$ne" : ... }
+ * is replaced by { "$not" : { "$eq" : ... } }
+ */
++ (NSDictionary *)replaceWithLonghand:(NSDictionary *)predicate
+{
+    NSMutableDictionary *accumulator = [NSMutableDictionary dictionary];
+    
+    if (!predicate || [predicate count] == 0) {
+        return predicate;
+    }
+    
+    NSString *operator = predicate.allKeys[0];
+    NSObject *subPredicate = predicate[operator];
+    if ([subPredicate isKindOfClass:[NSDictionary class]]) {
```

Comment:
That is what we are doing here.  I updated the block to match your suggestion in https://github.com/cloudant/CloudantQueryObjc/pull/50/files#r26143670

now it reads as:

```objc
    if (!predicate || [predicate count] == 0) {
        return predicate;
    }

    NSString *operator = predicate.allKeys[0];
    NSObject *subPredicate = predicate[operator];

    if ([subPredicate isKindOfClass:[NSDictionary class]]) {
       // Recurse down nested predicates, like { $not: { $not: { $ne: "blah" } } }
       return @{ operator: [CDTQQueryValidator replaceWithLonghand:(NSDictionary *)subPredicate] };
    } else if ([CDTQQueryValidator negatedShortHand][operator]) {
       // We got to the end and found an expandable operator, { $ne: "blah" }
       return @{ NOT: @{ [CDTQQueryValidator negatedShortHand][operator] : subPredicate } };
    } else {
       // We got to the end and found an normal operator, { $eq: "blah" }
       return @{ operator: subPredicate };
    }
```